### ISSUE TITLE
Few improvements on sct_run_batch

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#2726]
The following features would be nice:
- display header when running the command, as done in other SCT functions:
~~~
--
Spinal Cord Toolbox (VERSION-COMMIT)
~~~
- Create log file, which will include:
  - command ran
  - runtime
  - terminal output
- Send email once command has terminated (with output status)

**Comment by cfhammill** [comment]
Great ideas.

- added `sct_init()` at the top of the script
- in the process of adding the email feature.
- I'm a bit uncertain of the log file. I can definitely print either the full command string or just the task file. Do we want to add an extra flag for a general log when the user can run 
        ```
        sct_run_batch.py ... > terminal_log.txt
        ```
   or 
       ```
       sct_run_batch.py ... | tee terminal_log.txt
       ```

**Comment by jcohenadad** [comment]
i like the idea of an extra flag (many users are not familiar with `tee`)

**Comment by cfhammill** [comment]
Fair enough, pretty easy to add.

**Comment by cfhammill** [comment]
So I tried using `apprise` to send myself a gmail email, and it didn't work, it told me I needed to enable "less secure" app permissions for my account. I wasn't sure if I needed that to use smtp directly so I fixed up the `send_email` code but unfortunately that required the same thing. I can see an argument with sticking to the old smtp code, it doesn't require the user to learn the `apprise` syntax, and it might be easier to wrap with `getpass` so the user doesn't need to pass secrets on the command line (particularly if we wanted to dump the full command string to the logs). 

@jcohenadad or @zougloub do either of you feel strongly in favour of one approach or the other?

**Comment by jcohenadad** [comment]
I don’t feel strongly about a lot of things (including this ;-) )

**Comment by jcohenadad** [comment]
Another feature that would be nice (i just happen to need it right now): have the possibility to include a list of subjects to include/exclude. This is similar to the `-include` flag, but this flag currently does a regex filtering. In some cases it is not possible, and just listing subjects would be a good alternative.

so, one way to do it, in order to avoid adding flags (could be confusing), would be to add the possibility to include a "config.json" (for track record). We've already discussed that in the previous PR, now would be a good time to add that feature i guess. What do you think @cfhammill ?

**Comment by cfhammill** [comment]
Ok, I think I'm going to use `apprise` for notifications. It's a bit more complicated, but being able to do sms notifications was something I thought would be useful anyway. I'll need to think about how to ensure people aren't leaving passwords/tokens in their `.bash_history` and in the program logs.

For subject inclusion and exclusion lists, I guess collapsing the subject list to an exact matching regex is a bit heavy handed (and probably too much to ask users for). I'll definitely add this feature.

And yes, I think now would be a good time to add a `config.json`.

**Comment by jcohenadad** [comment]
> For subject inclusion and exclusion lists, I guess collapsing the subject list to an exact matching regex is a bit heavy handed (and probably too much to ask users for). I'll definitely add this feature.

cool, so what do you think would be a good approach? replacing the current regex feature by a list of subjects (separated by space)? We would need to be able to input a shell variable (e.g. `(sub1 sub2 sub3)`, so users can batch their processing. Also, i guess it would be compatible with a `config.json` approach (i.e. in the json file, there would be a field for "include", that would take a bracket-list of strings representing each subject)

**Comment by cfhammill** [comment]
I've already added a proposal for the inclusion and exclusion, I'll push the commit so we can discuss, but in brief I added two more ways to do inclusion/exclusion. So now we'd have:

1. `-include`, the regex version
2. `-include-list`, space separated list of subjects
3. `-include-file`, a single text file with one line per subject. 

It's a bit more repetitive than trying to guess on the command line which the user wants.
